pkg/pgn/parser.go

```diff
@@ -15,6 +16,43 @@ import (
 	"github.com/inhies/go-bytesize"
 )
 
+func PathParser(pgn_path string) ([]string, error) {
+	var files []string
+	fileTypes := []string{"pgn", "zst", "bz2"}
+	paths := strings.Split(pgn_path, ",")
+
+	for _, path := range paths {
+		file, err := os.Open(path)
```

Comment:
always close files. either by defer which will close file on function return or when you are done with it.
defer close should work fine here